testing/python/raises.py

```diff
@@ -19,6 +19,10 @@ def test_raises_function(self):
         excinfo = pytest.raises(ValueError, int, "hello")
         assert "invalid literal" in str(excinfo.value)
 
+    def test_raises_does_not_allow_empty_tuple(self):
+        with pytest.raises(ValueError):
```

Comment:
Let's match the message partially to ensure we don't regress this by accident:

```suggestion
        with pytest.raises(ValueError, match="Expected an exception type or"):
```

Reply:
Thanks @nicoddemus and @Zac-HD for the feedbacks.
@nicoddemus, I wasn't too sure about the following in your comment on the error message:
```python
"Expected an exception type or a tuple a tuple of exceptions types, but got {:r}"
                                                                            ^^^^
```
I used f-strings instead. Hope it's ok, and passes the tests.

Reply:
not sure why the this pre-commit (https://results.pre-commit.ci/run/github/37489525/1652147464.KX3rmadkSXKYLprqHEnP1A) failed. 
it's complaining about a test case I added to test for `expected_exception=None`. 
```
testing/python/raises.py:24: error: No overload variant of "raises" matches argument type "None"  [call-overload]
```
The test passes when I run the tests locally. Should I remove that test case?

Reply:
This is actually kinda nice - `mypy` is *also* warning us that `None` is invalid here.  Since I do want the runtime check to give us a nice error message, I've just pushed a `type: ignore` marker and explanatory comment.